src/mainFilesSidebarLoader.js

```diff
@@ -46,9 +45,45 @@ const isEnabled = function(fileInfo) {
 	return false
 }
 
+// It might be enough to keep the instance only in the Tab object itself,
+// without using a shared variable that can be destroyed if a new tab is
+// mounted and the previous one was not destroyed yet, as the tabs seem to
+// always be properly destroyed. However, this is how it is done for tabs in
+// server, so it is done here too just to be safe.
+let tabInstance = null
+
 window.addEventListener('DOMContentLoaded', () => {
 	if (OCA.Files && OCA.Files.Sidebar) {
 		OCA.Files.Sidebar.registerSecondaryView(new FilesSidebarCallView())
-		OCA.Files.Sidebar.registerTab(new OCA.Files.Sidebar.Tab('tab-chat', FilesSidebarTab, isEnabled))
+		OCA.Files.Sidebar.registerTab(new OCA.Files.Sidebar.Tab({
+			id: 'chat',
+			name: t('spreed', 'Chat'),
+			icon: 'icon-talk',
+			enabled: isEnabled,
+
+			async mount(el, fileInfo, context) {
+				if (tabInstance) {
+					tabInstance.$destroy()
+				}
+
+				// Dirty hack to force the style on parent component
+				const tabChat = document.querySelector('#tab-chat')
```

Comment:
no proper way ? can we instead set a class on the parent that we can style from here ?

Reply:
If I am not mistaken there have been some changes in the Files app sidebar related to this, I would need to check again.